static/app/views/performance/transactionSummary/teamKeyTransactionButton.tsx

```diff
@@ -38,152 +33,76 @@ class TitleButton extends Component<TitleProps> {
-    let counts: Map<string, number> = new Map();
-
-    let error: string | null = null;
-
-    try {
-      const teams = await fetchTeamKeyTransactions(
-        api,
-        organization.slug,
-        ['myteams'],
-        [String(project)]
-      );
-
-      keyedTeams = new Set(
-        teams
-          .filter(({keyed}) =>
-            keyed.find(
-              ({project_id, transaction}) =>
-                project_id === String(project) && transaction === transactionName
-            )
-          )
-          .map(({team}) => team)
-      );
-      counts = new Map(teams.map(({team, count}) => [team, count]));
-    } catch (err) {
-      error = err.responseJSON?.detail ?? t('Error fetching team key transactions');
-    }
-
-    this.setState({
-      isLoading: false,
-      keyFetchID: undefined,
-      error,
-      keyedTeams,
-      counts,
-    });
-  }
-
-  handleToggleKeyTransaction = async (
-    isKey: boolean,
-    teamIds: string[],
-    counts: Map<string, number>,
-    keyedTeams: Set<string>
-  ) => {
-    const {api, organization, project, transactionName} = this.props;
-    try {
-      await toggleKeyTransaction(
-        api,
-        isKey,
-        organization.slug,
-        [project],
-        transactionName,
-        teamIds
-      );
-      this.setState({
-        counts,
-        keyedTeams,
-      });
-    } catch (err) {
-      this.setState({
-        error: err.responseJSON?.detail ?? null,
-      });
-    }
+type Props = BaseProps &
+  TeamKeyTransactionManager.TeamKeyTransactionManagerChildrenProps & {
+    project: number;
   };
 
-  render() {
-    const {isLoading, error, counts, keyedTeams} = this.state;
-    return (
-      <TeamKeyTransaction
-        isLoading={isLoading}
-        error={error}
-        counts={counts}
-        keyedTeams={keyedTeams}
-        handleToggleKeyTransaction={this.handleToggleKeyTransaction}
-        title={TitleButton}
-        {...this.props}
-      />
-    );
-  }
+function TeamKeyTransactionButton({
+  counts,
+  getKeyedTeams,
+  project,
+  transactionName,
+  ...props
+}: Props) {
+  const keyedTeams = getKeyedTeams(String(project), transactionName);
+  return (
+    <TeamKeyTransactionComponent
+      counts={counts}
+      keyedTeams={keyedTeams}
+      title={TitleButton}
+      project={project}
+      transactionName={transactionName}
+      {...props}
+    />
+  );
 }
 
 type WrapperProps = BaseProps & {
   eventView: EventView;
 };
 
-function TeamKeyTransactionButtonWrapper({eventView, teams, ...props}: WrapperProps) {
+function TeamKeyTransactionButtonWrapper({
+  eventView,
+  organization,
+  teams,
+  ...props
+}: WrapperProps) {
   if (eventView.project.length !== 1) {
     return <TitleButton disabled keyedTeamsCount={0} />;
   }
 
+  const projectId = eventView.project[0];
   const userTeams = teams.filter(({isMember}) => isMember);
+
   return (
-    <TeamKeyTransactionButton
+    <TeamKeyTransactionManager.Provider
```

Comment:
We might want to lift this higher to be for all of performance later when you get to doing the landing, so that we only have the single state (we use key transaction on most of the performance pages anyway).

Reply:
So this is here because it's only for the team key transaction button on the transaction summary page. The star for the landing page will have its own component, with a lot of the code being shared with this one.

Reply:
It's own provider?  I'm saying the provider for whether a transaction is a key transaction could be lifted to cover all of performance so we don't need to make a new request every time you enter a transaction page.

Reply:
Ahh I see what you mean. We want to keep the response light for the transaction summary page so it we have to remake the request for each transaction summary page anyways. But I can revisit this and see if there's some other improvement to make here.